channel.go

```diff
@@ -402,7 +414,15 @@ func (ch *Channel) ListenAndServe(hostPort string) error {
 		return errAlreadyListening
 	}
 
-	l, err := net.Listen("tcp", hostPort)
+	var l net.Listener
+	var err error
+	if ch.enableMPTCP {
+		lc := &net.ListenConfig{}
+		lc.SetMultipathTCP(true)
+		l, err = lc.Listen(context.Background(), "tcp", hostPort)
+	} else {
+		l, err = net.Listen("tcp", hostPort)
+	}
```

Comment:
First, I can confirm the correctness of existing code. The [source code](https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/net/dial.go;l=432):
```lang=go
func Listen(network, address string) (Listener, error) {
	var lc ListenConfig
	return lc.Listen(context.Background(), network, address)
}
```
We can simplify this block and skip the if/else check. When the option is false, it is safe to set it as well.
```lang=go
lc := &net.ListenConfig{}
lc.SetMultipathTCP(ch.enableMPTCP) 
l, err := lc.Listen(context.Background(), "tcp", hostPort)
```